source/tasks/CreateOctopusRelease/CreateOctopusReleaseV6/task.json

```diff
@@ -0,0 +1,137 @@
+    "version": {
+        "Major": 6,
+        "Minor": 0,
+        "Patch": 0
+    },
+    "demands": [ ],
+    "minimumAgentVersion": "2.144.0",
+    "groups": [
+        {
+            "name": "versionControl",
+            "displayName": "Version Control",
+            "isExpanded": false
+        },
+        {
+            "name": "additional",
+            "displayName": "Additional Options",
+            "isExpanded": false
+        }
+    ],
+    "inputs": [
+        {
+            "name": "OctoConnectedServiceName",
+            "type": "connectedService:OctopusEndpoint",
+            "label": "Octopus Deploy Server",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Octopus Deploy server connection"
+        },
+        {
+            "name": "Space",
+            "type": "string",
+            "label": "Space",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "The space within Octopus. This must be the name of the space, not the id."
+        },
+        {
+            "name": "Project",
+            "type": "string",
+            "label": "Project",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "The project within Octopus. This must be the name of the project, not the id."
+        },
+        {
+            "name": "ReleaseNumber",
+            "type": "string",
+            "label": "Release Number",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "The number to use for this release. You can leave this blank if the release number is calculated by Octopus."
+        },
+        {
+            "name": "Channel",
+            "type": "string",
+            "label": "Channel",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "The [channel](https://g.octopushq.com/Channels) to use for the release. This must be the name of the channel, not the id."
+        },
+        {
+            "name": "DefaultPackageVersion",
+            "type": "string",
+            "label": "Default Package Version",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Set this to provide a default package version to use for all packages on all steps. Can be used in conjunction with the Packages field, which can be used to override versions for specific packages."
+        },
+        {
+            "name": "Packages",
+            "type": "multiLine",
+            "label": "Packages",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "A multi-line list of version numbers to use for a package in the release. Format: StepName:Version or PackageID:Version or StepName:PackageName:Version. StepName, PackageID, and PackageName can be replaced with an asterisk ('*'). An asterisk will be assumed for StepName, PackageID, or PackageName if they are omitted."
```

Comment:
Is there a difference between `PackageId` and `PackageName` here? It feels confusing

Reply:
There is, and yes it is. PackageID is what it sounds like, PackageName is the package reference name on the step, which is a far more confusing thing. For default packages (Deploy a Package step) it will be null/empty, for "additional packages" it will default to the PackageID when the reference is being added but the user can override it to whatever they want. It's there for some edge case where you can have the same PackageID referenced more than once on a step (to allow setting different versions for them during release creation. My understanding it's a use case where customer want V and V-1 to they can do some delta work on the packages)